ssh/tailssh/tailssh.go

```diff
@@ -1517,8 +1522,58 @@ func (ss *sshSession) startNewRecording() (_ *recording, err error) {
 			ss.logf("recording: error starting recording (failing open): %v", err)
 			return nil, nil
 		}
+
+		recorderDisappear := make(chan struct{})
+		go func() {
+			last := attempts[len(attempts)-1]
+
+			var mask ipn.NotifyWatchOpt
+			// identify the recorder node key by its tailscale IP
+			status := ss.conn.srv.lb.Status()
+			var recorderNodeID key.NodePublic
+			if status != nil && len(status.Peers()) > 0 {
+			outer:
+				for nodeID, peer := range status.Peer {
+					for _, ip := range peer.TailscaleIPs {
+						if ip == last.Recorder.Addr() {
+							recorderNodeID = nodeID
+							break outer
+						}
+					}
+				}
+			}
+			// if we can't find the recorder error out.
+			if recorderNodeID.IsZero() {
+				ss.logf("unable to find recorder node in status peer list")
+				return
+			}
+
+			// watch for IPN updates to observe if the recorder node ever goes offline.
+			ss.conn.srv.lb.WatchNotificationsAs(context.Background(), nil, mask, func() {}, func(roNotify *ipn.Notify) bool {
+				var recorderPeer tailcfg.NodeView
+				for _, peer := range roNotify.NetMap.Peers {
+					if peer.Key() == recorderNodeID {
+						recorderPeer = peer
+						break
+					}
+				}
+
+				if !recorderPeer.Valid() || !*recorderPeer.Online() {
+					ss.logf("previously selected recorder node is now offline or invalid; terminating session")
+					recorderDisappear <- struct{}{}
```

Comment:
to avoid a goroutine leak if `errChan` fires first:
```suggestion
					close(recorderDisappear)
```